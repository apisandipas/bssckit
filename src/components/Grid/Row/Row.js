import styled, { css } from 'styled-components';
import { ifProp } from 'styled-tools';
import defaultTheme from '../../../theme';
import { Div } from '../../Utilities/base';
import { Col } from '../Col';

const Row = styled(Div)`
  display: flex;
  flex-wrap: wrap;
  margin-right: -${props => props.theme.grid.gutterWidth / 2}px;
  margin-left: -${props => props.theme.grid.gutterWidth / 2}px;

  ${ifProp(
    'noGutters',
    css`
      margin-right: 0;
      margin-left: 0;

      > ${Col} {
        margin-right: 0;
        margin-left: 0;
      }
    `
  )}
`;

Row.defaultProps = {
  theme: defaultTheme,
};

export default Row;
