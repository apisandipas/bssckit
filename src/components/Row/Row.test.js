import React from 'react';
import { axe } from 'jest-axe';
import { render, cleanup } from '@testing-library/react';
import 'jest-styled-components';
import Row from './Row';

afterEach(cleanup);

describe('Row component', () => {
  it('matched Row styled', async () => {
    const { container } = render(<Row />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('display', 'flex');
    expect(container.firstChild).toHaveStyleRule('margin-left', '-15px');
    expect(container.firstChild).toHaveStyleRule('margin-right', '-15px');
    expect(container.firstChild).toHaveStyleRule('flex-wrap', 'wrap');
  });
});
