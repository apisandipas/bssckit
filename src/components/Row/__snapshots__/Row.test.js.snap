// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row component matched Row styled 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-right: -15px;
  margin-left: -15px;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<div
  class="sc-fznyAO c0"
/>
`;
